etcdserver/api/v2http/client_auth_test.go

```diff
@@ -361,6 +370,62 @@ func TestAuthFlow(t *testing.T) {
 	}
 }
 
+func (sh *authHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	sh.baseUsers(w, r)
+}
+
+func TestIssue5246(t *testing.T) {
+	sh := &authHandler{
+		sec: &mockAuthStore{
+			users: map[string]*auth.User{
+				"root": {
+					User:  "root",
+					Roles: []string{"root", "foo"},
+				},
+			},
+			roles: map[string]*auth.Role{
+				"root": {
+					Role: "root",
+				},
+			},
+		},
+		cluster: &fakeCluster{id: 1},
+	}
+	srv := httptest.NewServer(sh)
+	defer srv.Close()
+
+	req, err := http.NewRequest("GET", "", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.URL, err = url.Parse(srv.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	cli := http.DefaultClient
+	resp, err := cli.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer resp.Body.Close()
+
+	var uc usersCollections
+	if err := json.NewDecoder(resp.Body).Decode(&uc); err != nil {
+		t.Fatal(err)
+	}
+	for _, u := range uc.Users {
+		if u.User == "root" {
```

Comment:
this test does not ensure user root exist. it only fails if `root` exists and `foo` exists. it should also fail if `root` does not exist.